src/backend/web/graphql/index.js

```diff
@@ -113,37 +113,56 @@ module.exports.resolvers = {
      * @return Array of 'perPage' number of Post objects
      */
     getPosts: async (parent, { filter, page, perPage }) => {
+      let postIds;
+      const numOfPosts = await getPostsCount();
       const prPage = perPage > maxPostsPerPage ? maxPostsPerPage : perPage;
 
-      const numOFPosts = await getPostsCount();
-      const first = page * prPage;
-
-      if (first < numOFPosts) {
-        const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
-        const postIds = await getPosts(first, last);
-
+      // For Date Filter
+      if (filter) {
+        if (filter.fromDate || filter.toDate) {
+          const fromDate = filter.fromDate ? filter.fromDate : new Date();
+          const toDate = filter.toDate ? filter.toDate : new Date();
+          postIds = await getPostsByDate(fromDate, toDate); // Get all results within the date
+          // check if # of filtered results are less than max results allowed per page.
+          if (postIds.length < prPage) {
+            return Promise.all(postIds.map(id => Post.byId(id))); // return all the posts' info
```

Comment:
```js
return Promise.all(postIds.map(Post.byId));
```

Reply:
You can do that..?

Reply:
```js
// v1
map(function(id) { return Post.byId(id) })

//v2
map(id => Post.byId(id));

// v3
const fn = id => Post.byId(id);
map(fn)

// v4
map(Post.byId)
```

Because `map` takes a function that accepts an `id` argument and transforms it into a Post, and since `Post.byId` is a function that accepts an `id` argument and transforms it into a Post, you can use it directly without calling it via an outer function.

Reply:
TIL